controller/src/main/java/org/jboss/as/controller/registry/NotificationHandlerRegistration.java

```diff
@@ -44,38 +43,7 @@
  *
  * @author <a href="http://jmesnil.net/">Jeff Mesnil</a> (c) 2014 Red Hat inc.
  */
-public interface NotificationHandlerRegistration {
-
-    /**
-     * Special path address to register a notification handler for <em>any</em> source.
-     *
-     * A handler registered with this address will receive <em>all</em> notifications emitted by <em>any</em> source.
-     * It is advised to use a suitable {@code NotificationFilter} to constrain the received notifications (e.g. by their types).
-     */
-    PathAddress ANY_ADDRESS = PathAddress.EMPTY_ADDRESS;
-
-    /**
-     * Register the given NotificationHandler to receive notifications emitted by the resource at the given source address.
-     * The {@link org.jboss.as.controller.notification.NotificationHandler#handleNotification(org.jboss.as.controller.notification.Notification)} method will only be called on the registered handler if the filter's {@link org.jboss.as.controller.notification.NotificationFilter#isNotificationEnabled(org.jboss.as.controller.notification.Notification)}
-     * returns {@code true} for the given notification.
-     * <br />
-     *
-     * @param source the path address of the resource that emit notifications.
-     * @param handler the notification handler
-     * @param filter the notification filter. Use {@link org.jboss.as.controller.notification.NotificationFilter#ALL} to let the handler always handle notifications
-     */
-    void registerNotificationHandler(PathAddress source, NotificationHandler handler, NotificationFilter filter);
-
-    /**
-     * Unregister the given NotificationHandler to stop receiving notifications emitted by the resource at the given source address.
-     *
-     * The source, handler and filter must match the values that were used during registration to be effectively unregistered.
-     *
-     * @param source the path address of the resource that emit notifications.
-     * @param handler the notification handler
-     * @param filter the notification filter
-     */
-    void unregisterNotificationHandler(PathAddress source, NotificationHandler handler, NotificationFilter filter);
+public interface NotificationHandlerRegistration extends NotificationHandlerRegistry {
```

Comment:
Note: this interface doesn't need to be a subinterface; it could just expose the same methods. The users of NotificationHandlerRegistration are distinct from the users of NotificationHandlerRegistry. I did it this way because the implementation was simpler.

The NotificationHandlerRegistration users are internal to the kernel so we have freedom to change things in the future; e.g. to separate the interfaces.